src/main/java/seedu/address/model/module/Name.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.module;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Module's name in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
+ */
+public class Name {
```

Comment:
Perhaps it is better to change this to "ModuleName" to avoid confusion?

Reply:
I agree, might be mistaken for Name in Person